code/game/objects/structures/window.dm

```diff
@@ -276,7 +264,26 @@
 		return
 	playsound(src, "shatter", 70, 1)
 	var/turf/T = loc
-	for(var/obj/item/I in storeditems)
+	var/produced_items = list()
+
+	if(!(flags & NODECONSTRUCT))
+		var/shards = 1
+		if(fulltile)
+			shards++
+		var/rods = 0
+		if(reinf)
+			rods++
+			if(fulltile)
+				rods++
+
+		for(var/i in 1 to shards)
+			produced_items += new /obj/item/weapon/shard(src)
+		if(rods)
+			produced_items += new /obj/item/stack/rods(src, rods)
```

Comment:
The stored objects are not hard deleted when the window is destroyed, it's garbage collected and if there's already a lot of lag the system delays their deletion (thanks to qdel()). There is no such thing happening when creating new items.

Tbh I have no idea how much we save by using this method, you'd have to ask @hornygranny to know precisely.